skyportal/handlers/api/comment.py

```diff
@@ -129,24 +152,33 @@ def put(self, comment_id):
               application/json:
                 schema: Error
         """
-        c = Comment.query.get(comment_id)
+        c = Comment.get_if_owned_by(comment_id, self.current_user)
         if c is None:
             return self.error('Invalid comment ID.')
-        # Ensure user/token has access to parent source
-        _ = Source.get_if_owned_by(c.obj_id, self.current_user)
 
         data = self.get_json()
+        group_ids = data.pop("group_ids", None)
         data['id'] = comment_id
 
         schema = Comment.__schema__()
         try:
-            schema.load(data)
+            schema.load(data, partial=True)
         except ValidationError as e:
             return self.error('Invalid/missing parameters: '
                               f'{e.normalized_messages()}')
-
+        DBSession().flush()
+        if group_ids is not None:
```

Comment:
General question: for most objects, can we use the same machinery for creation and put?